### DISCUSSION TITLE
User account privilege elevation (again)

### DISCUSSION THREAD

**Discussion by papadeltasierra** [actions/virtual-environments#8029] [General questions] [answered]
I have an installer that does not run in a Windows/Powershell workflow because the installed wants UAC (i.e. pops up the "Allow changes..." popup if run on a regular Windows command prompt).  I found https://github.com/actions/runner-images/discussions/6557 and this suggested setting the `ConsentPromptBehaviorAdmin` item but I have tried that and still the installer does not run.

Examples of what I have tried are this:
```
Set-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System" -Name "ConsentPromptBehaviorAdmin" -Value 00000000 -Force
# then...
'& ${myInstaller...}'
# Or...
Start-Process -Verb RunAs -FilePath $myInstaller
```
Anything else I should be doing?  I have seen someone question whether the runner needs to be rebooted for the change t take affect - is this even possible?

Thanks for any and all suggestions.

**Comment by ilia-shipitsin** [comment]
I tried `Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System" -Name "ConsentPromptBehaviorAdmin"` for windows runner, it is 0 already.

UAC is disabled during image generation: https://github.com/actions/runner-images/blob/main/images/win/scripts/Installers/Initialize-VM.ps1#L28-L31

**Comment by vpolikarpov-akvelon** [comment]
Hey @papadeltasierra. What makes you think that UAC blocks execution? UAC is [already disabled ](https://github.com/actions/runner-images/blob/main/images/win/scripts/Installers/Initialize-VM.ps1#L28) in runner so there shouldn't by popup. Is it possible that installer needs something else to be executed in runner (e.g. flags for silent execution)? What software you are trying to install?

**Comment by papadeltasierra** [comment, marked as answer]
Oh, that's very interesting.  I'ts an installer that does have various parameters to stop pop-ups etc but I have set these and tested it locally.  But your info implies that maybe there is some other issue that is causing the installer to not run so I'll do a little more playing and try and figure out what is going on.
FWIW the installed is from here: http://wiki.telink-semi.cn/wiki/IDE-and-Tools/IDE-for-TLSR8-Chips/
When downloaded and then unzipped, there is an executable in the IDE drive named `TelinkSDKv1.3.1.exe` and that is what I am trying to run.

**Comment by papadeltasierra** [comment]
Umm, from what I can see, using the _call operator_ is not working.  As a test example I tried this:
```
- name: call me
  shell: pwsh
  run |
    $sorter="c:\windows\system32\sort.exe"
    & $sorter /?
```
I see no output at all, implying that the call operator doesn't appear to have run the sort command.

**Comment by papadeltasierra** [comment]
Success, and thank for your steer.  I managed to use this...
```
$proc=Start-Process -FilePath $installer -ArgumentList "<arguments here>" -RedirectStandardOutput <some file> ... -Wait -PassThru
```
